src/main/java/org/fcrepo/migration/handlers/BasicObjectVersionHandler.java

```diff
@@ -175,12 +162,210 @@ public void processObjectVersions(Iterable<ObjectVersionReference> versions) {
     }
 
     private boolean isDateProperty(String uri) {
-        return uri.equals("info:fedora/fedora-system:def/model#createdDate") || uri.equals("info:fedora/fedora-system:def/view#lastModifiedDate");
-
+        return uri.equals("info:fedora/fedora-system:def/model#createdDate") ||
+               uri.equals("info:fedora/fedora-system:def/view#lastModifiedDate") ||
+               uri.equals("http://www.loc.gov/premis/rdf/v1#hasDateCreatedByApplication") ||
+               uri.equals("http://www.loc.gov/premis/rdf/v1#hasEventDateTime");
     }
 
     private FedoraObject createObject(ObjectReference object) throws FedoraException {
         return repo.createObject(idMapper.mapObjectPath(object));
     }
 
+    private void updateObjectProperties(ObjectVersionReference version,
+                                        FedoraObject object,
+                                        QuadAcc triplesToRemove,
+                                        QuadDataAcc triplesToInsert) {
+        if (version.isLastVersion()) {
+            for (ObjectProperty p : version.getObjectProperties().listProperties()) {
+                String pred = p.getName();
+                String obj = p.getValue();
+
+                // Map dates and object state
+                if (pred.equals("info:fedora/fedora-system:def/model#createdDate")) {
+                    pred = "http://www.loc.gov/premis/rdf/v1#hasDateCreatedByApplication";
```

Comment:
Since we'll likely ultimately make these configurable, should we perhaps take the first step now and have our mapping encapsulated in a data structure or single method?